client/src/pages/Signup.jsx

```diff
@@ -0,0 +1,161 @@
+                      value: /^[A-Za-z]+$/i,
+                      message: 'Only letters allowed in last name',
+                    },
+                  })}
+                />
+              </Grid>
+              <Grid item xs={12}>
+                <TextField
+                  fullWidth
+                  required
+                  error={!!errors.email}
+                  helperText={errors.email?.message}
+                  variant="outlined"
+                  id="email"
+                  name="email"
+                  label="Email Address"
+                  autoComplete="email"
+                  inputRef={register({
+                    required: 'Email address is required',
+                    pattern: {
+                      value: /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/,
+                      message: 'Valid email format: xxxx@yyy.zzz',
+                    },
+                  })}
+                />
+              </Grid>
+              <Grid item xs={12}>
+                <TextField
+                  fullWidth
+                  required
+                  error={!!errors.password}
+                  helperText={errors.password?.message}
+                  variant="outlined"
+                  id="password"
+                  name="password"
+                  label="Password"
+                  type="password"
+                  inputRef={register({
+                    required: 'Password is required',
+                    minLength: { value: 6, message: 'Password must be at least six characters' },
+                  })}
+                />
+              </Grid>
+              <Grid item xs={12}>
+                <TextField
+                  fullWidth
+                  required
+                  error={!!errors.confirmPassword}
+                  helperText={errors.confirmPassword?.message}
+                  variant="outlined"
+                  id="confirmPassword"
+                  name="confirmPassword"
+                  label="Confirm Password"
+                  type="password"
+                  inputRef={register({
+                    validate: (val) => val === watch('password') || 'Passwords must match',
+                  })}
+                />
+              </Grid>
+              <Grid item xs={12} className={classes.linkText}>
+                <Typography component={Link} to="/login">
+                  Already have an account?
+                </Typography>
+              </Grid>
+            </Grid>
+            <Grid item container xs={12}>
+              <Grid item xs={4} />
+              <Grid item xs={4}>
+                <Button
+                  type="submit"
+                  variant="contained"
+                  fullWidth
+                  color="primary"
+                  className={classes.action}
+                >
+                  Create
+                </Button>
+              </Grid>
+              <Grid item xs={4} />
```

Comment:
I don't like these empty grids to center data - instead you should use the Grid flex-box abilities - you can add justify-content, etc. - read more about it here - https://material-ui.com/components/grid/